demo-2-migrate-otel.sh

```
@@ -0,0 +1,300 @@
+#   • Logs:    filelog (agent) + k8sobjects (gateway) → otlphttp/loki → Loki 3.x
+#   • Metrics: hostmetrics+kubeletstats (agent) + k8s_cluster (gateway)
+#              → prometheusremotewrite → Prometheus
+#   • Traces:  app OTLP → agent → gateway → Tempo
+#   • Prometheus scraping: prometheusreceiver with kubernetes_sd_configs
+#   • k8sattributes processor on every pipeline for cross-signal correlation
+#
+# Prereq: demo-1-before.sh has run and the cluster is in the "before" state.
+# Do NOT add --wait or --timeout to helm commands — causes failures on MacBook Air.
+# ─────────────────────────────────────────────────────────────────────────────
+set -uo pipefail
+cd "$(dirname "$0")"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)